troposphere/static/js/components/modals/instance/InstanceLaunchWizardModal.jsx

```diff
@@ -403,13 +415,35 @@ export default React.createClass({
     },
 
     onProjectCreateConfirm: function(name, description) {
-        this.viewBasic();
+        let sshKey = stores.SSHKeyStore.getAll();
+        if (sshKey) {
+            if (this.view != "IMAGE_VIEW" && sshKey.length === 0) {
+                this.view = "SSH_VIEW";
+                this.viewSSH();
```

Comment:
Nope. The reason why we have to do 421 and 422 is that we cannot change the state directly. We have to have a function that set the State. And `this.view` is just a variable that keeps track of what view that the user currently should be in and adjust the state later on.